### PR TITLE
Set endpoint description in Markdown

### PR DESCRIPTION
**Pull request by frodrigo**
Set endpoint description in Markdown #559
Map clearly Grape desc/detail to Swagger summary/description

### CODE REVIEW

File: spec/swagger_v2/api_swagger_v2_detail_spec.rb [hunk 1]
```diff
@@ -104,7 +106,7 @@ def app
     specify do
       expect(subject['paths']['/use_gfm_detail']['get']).to include('description')
       expect(subject['paths']['/use_gfm_detail']['get']['description']).to eql(
-        "<h1 id=\"this-returns-something\">This returns something</h1>\n<p># Burgers in Heaven</p>\n\n<blockquote>\n  <p>A burger doesn’t come for free</p>\n</blockquote>\n\n<p>If you want to reserve a burger in heaven, you have to do<br />\nsome crazy stuff on earth.</p>\n\n<pre><code>def do_good\nputs 'help people'\nend\n</code></pre>\n\n<ul>\n  <li><em>Will go to Heaven:</em> Probably</li>\n  <li><em>Will go to Hell:</em> Probably not</li>\n</ul>"
+        "> A burger doesn't come for free\n\nIf you want to reserve a burger in heaven, you have to do\nsome crazy stuff on earth.\n\n```\ndef do_good\nputs 'help people'\nend\n```\n\n* _Will go to Heaven:_ Probably\n* _Will go to Hell:_ Probably not\n"
```
Comment (LeFnord) [inline spec/swagger_v2/api_swagger_v2_detail_spec.rb]: please revert it, cause a MArkdown formmater is set (https://github.com/Mapotempo/grape-swagger/blob/a41387c14f667bd88fd63addf04f949e1d8e7673/spec/swagger_v2/api_swagger_v2_detail_spec.rb#L92), means it should be transformed to HTML

File: spec/swagger_v2/api_swagger_v2_detail_spec.rb [hunk 2]
```diff
@@ -142,7 +145,7 @@ def app
     specify do
       expect(subject['paths']['/use_gfm_rc_detail']['get']).to include('description')
       expect(subject['paths']['/use_gfm_rc_detail']['get']['description']).to eql(
-        "<h1>This returns something</h1>\n\n<p># Burgers in Heaven</p>\n\n<blockquote>\n<p>A burger doesn&#39;t come for free</p>\n</blockquote>\n\n<p>If you want to reserve a burger in heaven, you have to do\nsome crazy stuff on earth.</p>\n<pre class=\"highlight plaintext\"><code>def do_good\nputs 'help people'\nend\n</code></pre>\n<ul>\n<li><em>Will go to Heaven:</em> Probably</li>\n<li><em>Will go to Hell:</em> Probably not</li>\n</ul>"
+        "> A burger doesn't come for free\n\nIf you want to reserve a burger in heaven, you have to do\nsome crazy stuff on earth.\n\n```\ndef do_good\nputs 'help people'\nend\n```\n\n* _Will go to Heaven:_ Probably\n* _Will go to Hell:_ Probably not\n"
```
Comment (LeFnord) [inline spec/swagger_v2/api_swagger_v2_detail_spec.rb]: same here: https://github.com/Mapotempo/grape-swagger/blob/a41387c14f667bd88fd63addf04f949e1d8e7673/spec/swagger_v2/api_swagger_v2_detail_spec.rb#L131

**Comment by LeFnord** [comment]
thanks @frodrigo for adaptions … actual, the summary can be prtovided as plain text, as md, and as html rendered md, if the specific option was set 

yeap, it seems that the spec accept plain text and markdown, but not HTML, as you said, what did you think, should the markdown render stuff moved out?

**Comment by frodrigo** [comment]
I think for swagger 2 spec the HTML is no more required. And moreover tools using the spec expect markdown and convert it (e.g. swgger ui or swagger code gen deface the output, because they recive HTML and trait it as markdown).

**Comment by LeFnord** [comment]
would you remove all the markdown stuff then in this PR, or did you prefer to do it in a separate one?

**Comment by LeFnord** [comment]
@frodrigo … want to mörge it, the moving of Markdown out could be done in a separate PR, ok?

**Pull request merged**